src/interceptors/fetch_auth_token_from_api.js

```diff
@@ -11,23 +19,37 @@ import AddAuthTokenResponse from './add_auth_token_response';
 */
 export default class FetchAuthTokenFromApi {
   enter(ctx) {
-    const { tokenStore, authToken, endpointInterceptors, clientId } = ctx;
+    const { tokenStore, authToken, endpointInterceptors, clientId, inFlightAuthRequestStore } = ctx;
```

Comment:
Question: Now the storage object is stored in `ctx`. This is fine like this, but would it have worked if the promise was directly stored into `ctx`? So we'd just store the API call promise as `inFlightAuthRequest` attribute in `ctx`. Initially it would be set to `null` but would then hold the value during an ongoing request. Would that have worked too, or is there an upside, functionally or design-wise, to put the storage object between `ctx` and the promise?